### PR TITLE
Add multi-OS support to the CI tests

### PR DESCRIPTION
**Pull request by ssinad**
#### GitHub

- [x] I've given this PR a concise, self-descriptive, and meaningful title
- [x] I've linked relevant issues in the PR body
- [ ] I've applied [the relevant labels](https://www.neuro.polymtl.ca/software/contributing#pr_labels) to this PR
- [x] I've assigned a reviewer

#### PR contents

- [x] I've consulted [ivadomed's internal developer documentation](https://github.com/ivadomed/ivadomed/wiki) to ensure my contribution is in line with any relevant design decisions
- [ ] I've added [relevant tests](https://github.com/ivadomed/ivadomed/wiki/tests) for my contribution
- [ ] I've updated the [relevant documentation](https://github.com/ivadomed/ivadomed/wiki/documentation) for my changes, including argparse descriptions, docstrings, and ReadTheDocs tutorial pages

### CODE REVIEW

**Comment by dyt811** [comment]
@ssinad you can check my PR #757 that is just pretty much ready. You can review that one and it should work on most OS. Let me know if you have specific version of macOs you feel important to include in the test. 👍

File: .github/workflows/pre-commit.yml [hunk 1]
```diff
@@ -8,8 +8,14 @@ on:
...
 
 jobs:
+
   pre-commit-checks:
-    runs-on: 'ubuntu-latest'
+    strategy:
+      matrix:
+        os: [ "macos-latest", "windows-latest", "ubuntu-18.04", "ubuntu-20.04",  "macos-10.15"]
+        python-version: [3.6, 3.7, 3.8]
+
```
Comment (ssinad) [inline .github/workflows/pre-commit.yml:17]: @dyt811 I also added matrix parameters to the pre-commit.

**Comment by dyt811** [comment]
Hmmm... I think something is funky going on with macOS Python 3.8 and unit tests in particular... 
image

**Comment by ssinad** [comment]
> Hmmm... I think something is funky going on with macOS Python 3.8 and unit tests in particular...
> image

I was going to say.

**Comment by dyt811** [comment]
Oddly, I always thought new commits are suppose to auto cancel older/CI buid/queues. 

Judging by the queuing that is shown in this picture alone, we might get a stuck CI for the next day or so. 

image

Funny thing is... this seems to be very macOS specific as in your latest push, the Ubuntu/Windows CI are all done already but all MacOS are queued all way back in those commits.

**Comment by ssinad** [comment]
Closed in favour of #785.

**Pull request closed without merging**